service/gcsutils/gcstools/netnsConfig.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+// This utility moves a network interface into a network namespace and
+// configures it. The configuration is passed in as a JSON object
+// (marshalled prot.NetworkAdapter).  It is necessary to implement
+// this as a separate utility as in Go one does not have tight control
+// over which OS thread a given Go thread/routing executes but as can
+// only enter a namespace with a specific OS thread.
+//
+// Note, this logs to stdout so that the caller (gcs) can log the
+// output itself.
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/Microsoft/opengcs/service/gcs/prot"
+	log "github.com/Sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+func netnsConfigMain() {
+	if err := netnsConfig(); err != nil {
+		log.Error("netnsConfig returned: ", err)
```

Comment:
It's `logrus.Fatal` which calls `os.Exit`. The comments apply to the lines below them. It also confused me the first time I looked at it, I don't know why. You can also change log.Error to log.Fatal here if you want.